server/app.py

```diff
@@ -966,8 +967,24 @@ def payment_intent_succeeded(event):
 
 @celery.task(name="app.customer_subscription_deleted")
 def customer_subscription_deleted(event):
-    subscription = event["data"]["object"]
-    rdo = close_rdo(subscription["id"])
+    subscription = stripe.Subscription.retrieve(event["data"]["object"]["id"], expand=["customer"])
+    customer = subscription["customer"]
+    method = subscription["cancellation_details"].get('comment', 'Staff')
+    reason = subscription["cancellation_details"].get('reason')
+
+    contact = Contact.get(email=customer["email"])
+    rdo = close_rdo(subscription["id"], method=method, contact=contact)
+    text = f"{contact.name}'s ${rdo.amount}/{rdo.installment_period} donation was cancelled due to {reason}"
+    if reason == "cancellation_requested":
+        text += f" ({method})"
+
+    message = {
+        "text": text,
+        "channel": "#tech-test",
```

Comment:
**polish:** this would be cool to have as a config var. Maybe you were already scheming that

Reply:
Oh yeah, good point. Wouldn't be very helpful going there in prod :facepalm: Can fix.